geode-redis/src/main/java/org/apache/geode/redis/internal/executor/cluster/RedisPartitionResolver.java

```diff
@@ -15,16 +15,20 @@
 
 package org.apache.geode.redis.internal.executor.cluster;
 
+import static org.apache.geode.redis.internal.RegionProvider.REDIS_SLOTS;
+import static org.apache.geode.redis.internal.RegionProvider.REDIS_SLOTS_PER_BUCKET;
+
 import org.apache.geode.cache.EntryOperation;
 import org.apache.geode.cache.PartitionResolver;
-import org.apache.geode.redis.internal.data.ByteArrayWrapper;
 import org.apache.geode.redis.internal.data.RedisData;
+import org.apache.geode.redis.internal.data.RedisKey;
 
-public class RedisPartitionResolver implements PartitionResolver<ByteArrayWrapper, RedisData> {
+public class RedisPartitionResolver implements PartitionResolver<RedisKey, RedisData> {
 
   @Override
-  public Object getRoutingObject(EntryOperation<ByteArrayWrapper, RedisData> opDetails) {
-    return opDetails.getKey().getRoutingId();
+  public Object getRoutingObject(EntryOperation<RedisKey, RedisData> opDetails) {
+    // & (REDIS_SLOTS - 1) is equivalent to % REDIS_SLOTS but supposedly faster
```

Comment:
You made my brain explode :) Is this why we are restricting the number of buckets to a power of 2? That seems not worth doing just to use & instead of / here.

Reply:
No, it isn't the reason. The reason for powers of 2 is to maintain an even distribution of slots to buckets. If buckets is not a factor of slots then some buckets will hold more slots than others, leading to an imbalance of data.